src/overlays/actors/ovl_En_Rr/z_en_rr.c

```diff
@@ -21,51 +83,775 @@ const ActorInit En_Rr_InitVars = {
+        this->collider1.base.maskA &= ~2;
+        this->collider1.base.acFlags &= ~2;
+    }
+    func_8002E4B4(globalCtx, &this->actor, 20.0f, 30.0f, 20.0f, 7);
+    if (!this->stopScroll) {
+        Math_SmoothScaleMaxF(&this->segPhaseVel, this->segPhaseVelTarget, 1.0f, 50.0f);
+        Math_SmoothScaleMaxF(&this->segPulsePhaseDiff, 4.0f, 1.0f, 5.0f);
+        Math_SmoothScaleMaxF(&this->segWobblePhaseDiffX, this->segWobbleXTarget, 1.0f, 0.04f);
+        Math_SmoothScaleMaxF(&this->segWobblePhaseDiffZ, this->segWobbleZTarget, 1.0f, 0.01f);
+        Math_SmoothScaleMaxF(&this->pulseSize, this->pulseSizeTarget, 1.0f, 0.0015f);
+        Math_SmoothScaleMaxF(&this->wobbleSize, this->wobbleSizeTarget, 1.0f, 20.0f);
+        for (i = 0; i < 5; i++) {
+            Math_SmoothScaleMaxMinS(&this->bodySegs[i].rot.x, this->bodySegs[i].rotTarget.x, 5, this->segMoveRate * 1000.0f,
+                                    0);
+            Math_SmoothScaleMaxMinS(&this->bodySegs[i].rot.z, this->bodySegs[i].rotTarget.z, 5, this->segMoveRate * 1000.0f,
+                                    0);
+            Math_SmoothScaleMaxF(&this->bodySegs[i].scale.x, this->bodySegs[i].scaleTarget.x, 1.0f,
+                                 this->segMoveRate * 0.2f);
+            this->bodySegs[i].scale.z = this->bodySegs[i].scale.x;
+            Math_SmoothScaleMaxF(&this->bodySegs[i].height, this->bodySegs[i].heightTarget, 1.0f,
+                                 this->segMoveRate * 300.0f);
+        }
+        Math_SmoothScaleMaxF(&this->segMoveRate, 1.0f, 1.0f, 0.2f);
+    }
+}
+
+static Vec3f sEffectOffsets[] = {
+    { 25.0f, 0.0f, 0.0f },
+    { -25.0f, 0.0f, 0.0f },
+    { 0.0f, 0.0f, 25.0f },
+    { 0.0f, 0.0f, -25.0f },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE978C.s")
+void EnRr_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f zeroVec;
+    EnRr* this = THIS;
+    s32 i;
+    Mtx* segMtx = Graph_Alloc(globalCtx->state.gfxCtx, 4 * sizeof(Mtx));
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE9880.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_rr.c", 1478);
+    if (1) {}
+    func_80093D84(globalCtx->state.gfxCtx);
+    gSPSegment(POLY_XLU_DISP++, 0x0C, segMtx);
+    gSPSegment(POLY_XLU_DISP++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (this->scrollTimer * 0) & 0x7F,
+                                (this->scrollTimer * 0) & 0x3F, 32, 16, 1, (this->scrollTimer * 0) & 0x3F,
+                                (this->scrollTimer * -6) & 0x7F, 32, 16));
+    Matrix_Push();
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rr/func_80AE98F8.s")
+    Matrix_Scale((1.0f + this->bodySegs[RR_BASE].scaleMod.x) * this->bodySegs[RR_BASE].scale.x,
+                 (1.0f + this->bodySegs[RR_BASE].scaleMod.y) * this->bodySegs[RR_BASE].scale.y,
+                 (1.0f + this->bodySegs[RR_BASE].scaleMod.z) * this->bodySegs[RR_BASE].scale.z, 1);
```

Comment:
Relevant in other places
```suggestion
                 (1.0f + this->bodySegs[RR_BASE].scaleMod.z) * this->bodySegs[RR_BASE].scale.z, MTXMODE_APPLY);
```